sdks/java/core/src/main/java/org/apache/beam/sdk/util/ReduceFnRunner.java

```diff
@@ -211,7 +213,9 @@ public ReduceFnRunner(
       PipelineOptions options) {
     this.key = key;
     this.timerInternals = timerInternals;
-    this.paneInfoTracker = new PaneInfoTracker(timerInternals);
+    this.paneInfoTracker = GlobalWindows.INSTANCE.isCompatible(windowingStrategy.getWindowFn()) ?
```

Comment:
Ah ha! You already knew isCompatible is wrong headed for this. I figured it out the hard way.

In the end consistently using the 'compare maxTimestamp against GlobalWindow.INSTANCE.maxTimestamp' test seems the most robust to me. See what you think.